src/string.cr

```diff
@@ -2441,11 +2441,35 @@ class String
   # old_pond.split(3) # => ["Old", "pond", "a frog leaps in\n  water's sound\n"]
   # ```
   def split(limit : Int32? = nil)
+    ary = Array(String).new
+    split(limit) do |string|
+      ary << string
+    end
+    ary
+  end
+
+  # Splits the string after any ASCII whitespace character and yields each part to a block.
+  #
+  # If *limit* is present, up to *limit* new strings will be created,
+  # with the entire remainder added to the last string.
+  #
+  # ```
+  # ary = [] of String
+  # old_pond = "
+  #   Old pond
+  #   a frog leaps in
+  #   water's sound
+  # "
+  # old_pond.split { |s| ary << s }; ary    # => ["Old", "pond", "a", "frog", "leaps", "in", "water's", "sound"]
+  # old_pond.split(3) { |s| ary << s }; ary # => ["Old", "pond", "a frog leaps in\n  water's sound\n"]
```

Comment:
`ary` should probably cleared before each split. This applies to this example and other examples below.